api/migrations/20240711094931-2023-2024_classe_statusPhase1.js

```diff
@@ -0,0 +1,28 @@
+const ClasseModel = require("../src/models/cle/classe");
+const YoungModel = require("../src/models/young");
+const { STATUS_PHASE1_CLASSE, YOUNG_STATUS_PHASE1 } = require("snu-lib");
+
+module.exports = {
+  async up(db, client) {
+    const classes = await ClasseModel.find({ schoolYear: "2023-2024" });
+
+    const classIds = classes.map((classe) => classe._id);
+
+    const youngsDone = await YoungModel.find({ classeId: { $in: classIds }, statusPhase1: YOUNG_STATUS_PHASE1.DONE });
+
+    const doneClassIds = new Set(youngsDone.map((young) => young.classeId));
+
+    await ClasseModel.updateMany(
+      {
+        ligneId: { $exists: true },
+        _id: { $nin: Array.from(doneClassIds) },
+      },
+      { $set: { statusPhase1: STATUS_PHASE1_CLASSE.AFFECTED } },
+    );
+    await ClasseModel.updateMany({ _id: { $in: Array.from(doneClassIds) } }, { $set: { statusPhase1: STATUS_PHASE1_CLASSE.DONE } });
```

Comment:
```suggestion
    await ClasseModel.updateMany({ _id: { $in: Array.from(doneClassIds) } }, { statusPhase1: STATUS_PHASE1_CLASSE.DONE });
```
détail, il me semble que l'on n'est pas obligé de passer le `$set`